Userland/CMakeLists.txt

```diff
@@ -14,6 +14,7 @@ target_link_libraries(disasm LibX86)
 target_link_libraries(functrace LibDebug LibX86)
 target_link_libraries(html LibWeb)
 target_link_libraries(ht LibWeb)
+target_link_libraries(keymap LibKeyboard LibCore)
```

Comment:
No need to add `LibCore` here explicitly, all userland executables already link to it.

Reply:
Hi,
When i remove it, i get following error.

```
[0/1] Re-running CMake...
-- Configuring done
-- Generating done
-- Build files have been written to: /home/asliturk/git-repos/serenity/Build
[1/1] Linking CXX executable Userland/keymap
FAILED: Userland/keymap 
: && /home/asliturk/git-repos/serenity/Toolchain/Local/bin/i686-pc-serenity-g++  -Wall -Wextra -Werror -std=c++2a -fdiagnostics-color=always -Os -Wno-sized-deallocation -fno-sized-deallocation -fno-exceptions -fno-rtti -Wno-address-of-packed-member -Wundef -Wcast-qual -Wwrite-strings -Wimplicit-fallthrough -Wno-nonnull-compare -Wno-deprecated-copy -Wno-expansion-to-defined -DDEBUG -DSANITIZE_PTRS  --sysroot /home/asliturk/git-repos/serenity/Build/Root Userland/CMakeFiles/keymap.dir/keymap.cpp.o  -o Userland/keymap -L/home/asliturk/git-repos/serenity/Build/Libraries/LibC   -L/home/asliturk/git-repos/serenity/Build/Libraries/LibM   -L/home/asliturk/git-repos/serenity/Build/Libraries/LibPthread -Wl,-rpath,/home/asliturk/git-repos/serenity/Build/Libraries/LibC:/home/asliturk/git-repos/serenity/Build/Libraries/LibM:/home/asliturk/git-repos/serenity/Build/Libraries/LibPthread:  Libraries/LibCore/libcore.a  Libraries/LibKeyboard/libkeyboard.a  Libraries/LibC/libc.a  Libraries/LibC/libcrt0.a && :
/home/asliturk/git-repos/serenity/Toolchain/Local/lib/gcc/i686-pc-serenity/10.1.0/../../../../i686-pc-serenity/bin/ld: Libraries/LibKeyboard/libkeyboard.a(CharacterMapFile.cpp.o): in function `Keyboard::CharacterMapFile::load_from_file(AK::String const&)':
CharacterMapFile.cpp:(.text+0x3a9): undefined reference to `Core::File::File(AK::StringView const&, Core::Object*)'
/home/asliturk/git-repos/serenity/Toolchain/Local/lib/gcc/i686-pc-serenity/10.1.0/../../../../i686-pc-serenity/bin/ld: CharacterMapFile.cpp:(.text+0x3d1): undefined reference to `Core::File::open(Core::IODevice::OpenMode)'
/home/asliturk/git-repos/serenity/Toolchain/Local/lib/gcc/i686-pc-serenity/10.1.0/../../../../i686-pc-serenity/bin/ld: CharacterMapFile.cpp:(.text+0x41e): undefined reference to `Core::IODevice::error_string() const'
/home/asliturk/git-repos/serenity/Toolchain/Local/lib/gcc/i686-pc-serenity/10.1.0/../../../../i686-pc-serenity/bin/ld: CharacterMapFile.cpp:(.text+0x45e): undefined reference to `Core::IODevice::read_all()'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
 ~
```

Reply:
This sounds like LibKeyboard uses LibCore, but doesn't link to it properly.